dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -358,9 +358,13 @@ def allows_section_takeover(user)
     # OAuth providers do not necessarily provide student email addresses, so we
     # want to perform silent takeover on these accounts, but *only if* the
     # student hasn't made progress with the initial account
+    auth_options_emails = user.migrated? ?
+      user.authentication_options.select {|ao| !ao.hashed_email.blank? && ao.hashed_email} :
+      []
```

Comment:
Could this be simplified as 
`has_auth_email = user.migrated? && user.authentication_options.any? {|ao| ao.hashed_email.present?}`

Reply:
`present?` is a good solution here. I'll update this.